cw721-timelock/src/contract.rs

```diff
@@ -0,0 +1,272 @@
+    let contract = ADOContract::default();
+
+    let resp = contract.instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        &deps.querier,
+        info.clone(),
+        BaseInstantiateMsg {
+            ado_type: CONTRACT_NAME.to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            kernel_address: msg.kernel_address,
+            owner: msg.owner,
+        },
+    )?;
+
+    if let Some(authorized_token_addresses) = msg.authorized_token_addresses {
+        if !authorized_token_addresses.is_empty() {
+            ADOContract::default().permission_action(SEND_NFT_ACTION, deps.storage)?;
+        }
+
+        for token_address in authorized_token_addresses {
+            let addr = token_address.get_raw_address(&deps.as_ref())?;
+            ADOContract::set_permission(
+                deps.storage,
+                SEND_NFT_ACTION,
+                addr,
+                Permission::Whitelisted(None),
+            )?;
+        }
+    }
+
+    Ok(
+        resp
+        .add_attribute("method", "instantiate")
+        .add_attribute("owner", info.sender)
+    )
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ctx = ExecuteContext::new(deps, info, env);
+
+    match msg {
+        ExecuteMsg::AMPReceive(pkt) => {
+            ADOContract::default().execute_amp_receive(ctx, pkt, handle_execute)
+        }
+        _ => handle_execute(ctx, msg),
+    }
+}
+
+pub fn handle_execute(ctx: ExecuteContext, msg: ExecuteMsg) -> Result<Response, ContractError> {
+
+    match msg {
+        ExecuteMsg::ReceiveNft(msg) => handle_receive_cw721(ctx, msg),
+        ExecuteMsg::ClaimNft {
+            cw721_contract,
+            token_id,
+        } => execute_claim_cw721(ctx, cw721_contract, token_id),
+        _ => ADOContract::default().execute(ctx, msg),
+    }
+}
+
+fn handle_receive_cw721(ctx: ExecuteContext, msg: Cw721ReceiveMsg) -> Result<Response, ContractError> {
+
+    ADOContract::default().is_permissioned(
+        ctx.deps.storage,
+        ctx.env.clone(),
+        SEND_NFT_ACTION,
+        ctx.info.sender.clone(),
+    )?;
+
+    match from_json(&msg.msg)? {
+        Cw721HookMsg::TimelockNft {
+            lock_duration,
+            recipient,
+        } => execute_timelock_cw721(
+            ctx,
+            msg.sender,
+            msg.token_id,
+            lock_duration,
+            recipient,
+        ),
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn execute_timelock_cw721(
+    ctx: ExecuteContext,
+    _sender: String,
+    token_id: String,
+    lock_duration: MillisecondsDuration,
+    recipient: Recipient,
+) -> Result<Response<Empty>, ContractError> {
+    let ExecuteContext {
+        deps,
+        info,
+        env,
+        ..
+    } = ctx;
+
+    ensure!(
+        lock_duration.seconds() >= ONE_DAY,
```

Comment:
Any reason to choose `ONE_DAY` here?

Reply:
![image](https://github.com/andromedaprotocol/ado-database/assets/169449044/0b19f3e2-5538-4619-9197-b8840f2d962a)